Eatery/Controllers/BRB/BRBLoginViewController.swift

```diff
@@ -0,0 +1,338 @@
+        let loginAndActivityContainerView = UIView(frame: .zero)
+        
+        loginButton = UIButton(type: .system)
+        loginButton.setTitle("Login", for: .normal)
+        loginButton.setTitle("", for: .disabled)
+        loginButton.layer.cornerRadius = 8.0
+        loginButton.layer.masksToBounds = true
+        loginButton.contentEdgeInsets = UIEdgeInsets(top: 16, left: 8, bottom: 16, right: 8)
+        loginButton.backgroundColor = .eateryBlue
+        loginButton.setTitleColor(.white, for: .normal)
+        loginButton.titleLabel?.font = .preferredFont(forTextStyle: .headline)
+        loginButton.addTarget(self, action: #selector(loginButtonPressed(_:)), for: .touchUpInside)
+        loginAndActivityContainerView.addSubview(loginButton)
+        loginButton.snp.makeConstraints { make in
+            make.edges.equalToSuperview()
+        }
+        
+        activityIndicator = NVActivityIndicatorView(frame: .zero, type: .circleStrokeSpin, color: .white)
+        loginAndActivityContainerView.addSubview(activityIndicator)
+        activityIndicator.snp.makeConstraints { make in
+            make.center.equalToSuperview()
+            make.width.height.equalTo(22)
+        }
+        
+        stackView.addArrangedSubview(loginAndActivityContainerView)
+    }
+    
+    @objc private func privacyStatementButtonPressed(_ sender: UIButton) {
+        let privacyStatementViewController = BRBPrivacyStatementViewController()
+        navigationController?.pushViewController(privacyStatementViewController, animated: true)
+    }
+    
+    @objc private func saveMyLoginInfoSwitchToggled(_ sender: UISwitch) {
+        BRBAccountSettings.saveLoginInfo = saveLoginInfoSwitch.isOn
+        
+        if !saveLoginInfoSwitch.isOn {
+            BRBAccountSettings.removeKeychainLoginInfo()
+        }
+    }
+    
+    @objc private func loginButtonPressed(_ sender: UIButton) {
+        requestLoginIfPossible()
+    }
+    
+    private func requestLoginIfPossible() {
+        let netid = netidTextField.text?.lowercased() ?? ""
+        let password = passwordTextField.text ?? ""
+        
+        guard !netid.isEmpty else {
+            netidTextField.becomeFirstResponder()
+            return
+        }
+        
+        guard !password.isEmpty else {
+            passwordTextField.becomeFirstResponder()
+            return
+        }
+    
+        netidTextField.resignFirstResponder()
+        passwordTextField.resignFirstResponder()
+        
+        if BRBAccountSettings.saveLoginInfo {
+            let loginInfo = (netid: netid, password: password)
+            BRBAccountSettings.saveToKeychain(loginInfo: loginInfo)
+        }
+        
+        delegate?.loginViewController(self, didRequestLoginWithNetid: netid, password: password)
+    }
+    
+    func setShowErrorMessage(_ newValue: Bool, animated: Bool) {
+        let actions: () -> Void = { [weak self] in
+            self?.errorView.isCollapsed = !newValue
```

Comment:
can we unwrap self with a guard let?

Reply:
I think the better action here would just be to retain a strong reference to `self`. In the first case, the `UIViewPropertyAnimator` immediately falls out of scope, and in the second case the closure never even leaves the function scope